x/gamm/client/cli/tx.go

```diff
@@ -178,6 +181,115 @@ func NewStableSwapAdjustScalingFactorsCmd() *cobra.Command {
+		Long: strings.TrimSpace(`Submit a replace migration record proposal.
+
+Passing in poolIds separated by commas would be parsed automatically to pairs of migration record.
+Ex) 2,4,1,5 -> [(Balancer 2, CL 4), (Balancer 1, CL 5)]
+
+
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			content, err := parseReplaceMigrationRecordsArgsToContent(cmd)
+			if err != nil {
+				return err
+			}
+
+			from := clientCtx.GetFromAddress()
+
+			depositStr, err := cmd.Flags().GetString(govcli.FlagDeposit)
+			if err != nil {
+				return err
+			}
+			deposit, err := sdk.ParseCoinsNormalized(depositStr)
+			if err != nil {
+				return err
+			}
+
+			msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+			if err != nil {
+				return err
+			}
+
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	cmd.Flags().String(govcli.FlagTitle, "", "title of proposal")
+	cmd.Flags().String(govcli.FlagDescription, "", "description of proposal")
+	cmd.Flags().String(govcli.FlagDeposit, "", "deposit of proposal")
+	cmd.Flags().String(FlagMigrationRecords, "", "The migration records array")
+
+	return cmd
+}
+
+// NewCmdSubmitUpdateMigrationRecordsProposal implements a command handler for replace migration records proposal
+func NewCmdSubmitUpdateMigrationRecordsProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-migration-records-proposal [flags]",
+		Args:  cobra.ExactArgs(0),
+		Short: "Submit a update migration record proposal",
+		Long: strings.TrimSpace(`Submit a update migration record proposal.
+
+Passing in poolIds separated by commas would be parsed automatically to pairs of migration record.
+Ex) 2,4,1,5 -> [(Balancer 2, CL 4), (Balancer 1, CL 5)]
+
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			content, err := parseUpdateMigrationRecordsArgsToContent(cmd)
+			if err != nil {
+				return err
+			}
+
+			from := clientCtx.GetFromAddress()
+
+			depositStr, err := cmd.Flags().GetString(govcli.FlagDeposit)
+			if err != nil {
+				return err
+			}
+			deposit, err := sdk.ParseCoinsNormalized(depositStr)
+			if err != nil {
+				return err
+			}
+
+			msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+			if err != nil {
+				return err
+			}
+
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	cmd.Flags().String(govcli.FlagTitle, "", "title of proposal")
+	cmd.Flags().String(govcli.FlagDescription, "", "description of proposal")
+	cmd.Flags().String(govcli.FlagDeposit, "", "deposit of proposal")
+	cmd.Flags().String(FlagMigrationRecords, "", "The migration records array")
+
+	return cmd
+}
+
```

Comment:
Why are we not using osmocli for these?

Reply:
@czarcas7ic  Can osmocli handle specific flags and logics inside the RunE function as well?

Reply:
It should be possible to add custom flag parsers:
https://github.com/osmosis-labs/osmosis/blob/9743eaae0925eed56ff5e7305935a54266220c1c/x/poolmanager/client/cli/tx.go#L43-L45

Reply:
Resolving this due to issues mentioned below